cli/run_test.go

```diff
@@ -0,0 +1,176 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var dispatchBinary = filepath.Join("../build", runtime.GOOS, runtime.GOARCH, "dispatch")
+
+func TestRunCommand(t *testing.T) {
+	t.Run("Run with non-existent env file", func(t *testing.T) {
+		t.Parallel()
+
+		buff, msg, err := execRunCommand(&[]string{}, "run", "--env-file", "non-existent.env", "--", "echo", "hello")
+		if err != nil {
+			t.Fatalf(msg, err)
+		}
+
+		assert.Regexp(t, "Error: failed to load env file from .+: open .+: no such file or directory\n", buff.String())
```

Comment:
Could we test for exact matches to be more strict?

Reply:
Paths will be different for different platforms and users, but I can do them a little more strict with checking known parts of the path

Reply:
If you use absolute paths where the base comes from `t.TempDir()` you'll have a portable solution.

Reply:
improved it.

To add more context:
The first one is absolute path `<path to the parent dir of the dispatch project>/dispatch/cli/non-existent.env`. So I'm checking that the error prints understandable error here.